JDK-8279242: Reflection newInstance() error message when constructor has no access modifiers could use improvement

jddarcy opened the pull request:
Making the exception message friendlier to users.

dholmes-ora commented on src/java.base/share/classes/jdk/internal/reflect/Reflection.java:
```diff
@@ -378,11 +378,14 @@ public static IllegalAccessException newIllegalAccessException(Class<?> currentC
             msg += "a member of " + memberClass + memberSuffix +
-                    " with modifiers \"" + Modifier.toString(modifiers) + "\"";
+                (packageAccess ?
+                 " package access" :
```
Suggestion: "with package access".

jddarcy replied:
Updated as suggested.

dholmes-ora reviewed (approved):
Looks good.

I see Mandy has suggested the same wording in the JBS issue.

Thanks,
David

mlchung commented:
There is another `newIllegalAccessException(Class<?> memberClass, int modifiers)` method that also needs to be updated.

jddarcy commented:
> There is another `newIllegalAccessException(Class<?> memberClass, int modifiers)` method that also needs to be updated.

Updated accordingly and refactored slightly to pull out the string suffix construction into a new method.

mlchung reviewed (approved):
Looks good.

jddarcy commented:
/integrate
